app/vmalert/group_test.go

```diff
@@ -105,20 +107,32 @@ func TestUpdateWith(t *testing.T) {
 				{Record: "foo5"},
 			},
 		},
+		{
+			"update datasource type",
```

Comment:
hm, I was expecting this test to fail, because we don't update type in rule.UpdateWith method
but looks we don't even call it anymore... We always re-create rule instead. I guess, it is my mistake and I'll take a closer look later.

Reply:
Its called when `for` and `annotations` params changes, as far as i remember. In other cases, it generates new ruleID.